src/stapi_fastapi/models/order.py

```diff
@@ -1,15 +1,44 @@
-from typing import Literal
+from typing import Generic, Literal, TypeVar
 
 from geojson_pydantic import Feature, FeatureCollection
 from geojson_pydantic.geometries import Geometry
 from pydantic import BaseModel, ConfigDict, Field, StrictInt, StrictStr
 
+from stapi_fastapi.models.opportunity import OpportunityProperties
 from stapi_fastapi.models.shared import Link
 from stapi_fastapi.types.datetime_interval import DatetimeInterval
+from stapi_fastapi.types.filter import CQL2Filter
 
 
-class OrderProperties(BaseModel):
+class OrderParameters(BaseModel):
+    model_config = ConfigDict(extra="allow")
+
+
+ORP = TypeVar("ORP", bound=OrderParameters)
+OPP = TypeVar("OPP", bound=OpportunityProperties)
+
+
+class OrderRequest(BaseModel, Generic[ORP]):
```

Comment:
I didn't link this via inheritance with OpportunityRequest because they're different things, and I didn't think it was worth a few lines to share the model code between them